### PR TITLE
chore: start args in a configmap

### PR DESCRIPTION
**Pull request by whynowy**
With more starting args introduced, updating the deployment args becomes inconvenient. Change to support both configmap and args (recommend to use configmap, but args take precedence).

Use a single optional configmap for both controller and UX server.

### CODE REVIEW

**Comment by whynowy** [comment]
Do not merge, need to think through.

**Review by vigith — APPROVED** [review]

**Comment by whynowy** [comment]
This is good to review. @numaproj/numaflow-dev

File: config/base/shared-config/numaflow-cmd-params-config.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+metadata:
+  name: numaflow-cmd-params-config
+data:
+  ### Whether to run the controller and the UX server in namespaced scope, defaults to false.
+  # namespaced: "false"
+  #
+  ### The namespace that the controller and the UX server watch when "namespaced" is true.
+  # managed.namespace: numaflow-system
+  #
+  ### Where to disable leader election for the controller, defaults to false
```
Comment (KeranYang) [inline config/base/shared-config/numaflow-cmd-params-config.yaml:12]:
```suggestion
  ### Whether to disable leader election for the controller, defaults to false
```

File: docs/operations/installation.md [hunk 2]
```diff
@@ -74,11 +100,16 @@ To do managed namespace installation, besides `--namespaced`, add `--managed-nam
...
-```
-      name: NUMAFLOW_LEADER_ELECTION_DISABLED
-      value: "true"
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: numaflow-cmd-params-config
+data:
+  # Where to disable leader election for the controller, defaults to false
```
Comment (KeranYang) [inline docs/operations/installation.md:111]:
```suggestion
  # Whether to disable leader election for the controller, defaults to false
```

**Review by KeranYang — APPROVED** [review]

**Review by vigith — APPROVED** [review]

**Pull request merged**